dpctl/_sycl_device.pyx

```diff
@@ -157,6 +188,87 @@ cdef class SyclDevice(_SyclDevice):
             SyclDevice._init_helper(self, DRef)
             return 0
 
+    cdef _raise_sub_devices_creation_error(self, fname, errcode):
+        e = SubDeviceCreationError("Sub-devices were not created.")
+        e.fname = fname
+        e.code = errcode
+        raise e
```

Comment:
This function serves no purpose. It is deleted in #343 

Object `e` derives from `Exception`, which does not have attributes `fname` or `code`. Lines 193-194 dynamically add these attributes to the exception instance, but they go unused. 

Calls `self._raise_sub_devices_creation_error("DPCTLSubDeviceCreationError", -1)` have been replaced with `raise SubDeviceCreationError("Sub-devices were not created.")` in #343